### ISSUE TITLE
Google Sheets: Token has been expired or revoked.

### ISSUE THREAD

**Issue by gltched-usr** [formbricks/formbricks#4344]
### Issue Summary

Today I was checking on the google sheets on wondered why I only got 5 responses. When looking on formbricks I can see that there are 38 responses so far. When checking the logs I saw: `400: Token has been expired or revoked.`. Is there a way to renew and then resync the responses?

### Expected Behavior

Maybe when the token is about to expire notify an admin (maybe via mail) with instructions on how to renew.

### Your Environment

Self-hosted Formbricks

**Comment by Dhruwang** [comment]
@mattinannt , few months back we fixed this issue in this PR 
https://github.com/formbricks/formbricks/pull/2667

![Image](https://github.com/user-attachments/assets/7dcdcef3-f93c-479d-ac83-ca066dc4dff1)


We have this authorize function which returns an oauthClient and it runs every time we make a request to google sheet, and everytime it uses the refresh token to generate latest credentials and update the existing configuration

**Comment by Dhruwang** [comment]
Hey @gltched-usr , can you share the formbricks version that you are using ? 😊

**Comment by Dhruwang** [comment]
Hey @gltched-usr , does the above solution help ? 😊

**Comment by Dhruwang** [comment]
We updated our docs for this 😊

**Comment by sapnesh-nango** [comment]
Yeah testing mode app tokens from Google expire in [7 days](https://nango.dev/blog/google-oauth-invalid-grant-token-has-been-expired-or-revoked).
